From d096192f401cbbb75f6f6ffd7dcda3766a5505b9 Mon Sep 17 00:00:00 2001
Message-Id: <d096192f401cbbb75f6f6ffd7dcda3766a5505b9.1510090607.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 7 Nov 2017 22:11:47 +0100
Subject: [PATCH] meson: Make sure the entire wireless-security static lib is
 used

Otherwise ld will not link in the gresources, which contain no "needed"
(as far as ld can determine) symbols.
---
 src/connection-editor/meson.build |  5 +++--
 src/libnm-gtk/meson.build         |  5 +++--
 src/libnma/meson.build            |  5 +++--
 src/meson.build                   | 10 +++++++++-
 src/wireless-security/meson.build | 10 ----------
 5 files changed, 18 insertions(+), 17 deletions(-)

diff --git a/src/connection-editor/meson.build b/src/connection-editor/meson.build
index 990f1b4739ff6e4f..004303e48d21acf1 100644
--- a/src/connection-editor/meson.build
+++ b/src/connection-editor/meson.build
@@ -76,14 +76,14 @@ incs = [
   top_inc,
   utils_inc,
   src_inc,
-  shared_inc
+  shared_inc,
+  wireless_security_inc
 ]
 
 deps = [
   gtk_dep,
   libnm_dep,
   libnma_dep,
-  libwireless_security_libnm_dep,
   m_dep
 ]
 
@@ -117,6 +117,7 @@ executable(
   c_args: cflags,
   link_args: ldflags,
   link_depends: linker_script_ver,
+  link_whole: libwireless_security_libnm,
   install: true,
   install_dir: nma_bindir
 )
diff --git a/src/libnm-gtk/meson.build b/src/libnm-gtk/meson.build
index 9aaf21942cae8fe6..cc6785dadd04a85c 100644
--- a/src/libnm-gtk/meson.build
+++ b/src/libnm-gtk/meson.build
@@ -39,15 +39,15 @@ incs = [
   top_inc,
   shared_inc,
   src_inc,
-  libnma_inc
+  libnma_inc,
+  wireless_security_inc
 ]
 
 deps = [
   gtk_dep,
   gudev_dep,
   libnm_glib_dep,
   libutils_libnm_glib_dep,
-  libwireless_security_libnm_glib_dep
 ]
 
 cflags = [
@@ -77,6 +77,7 @@ libnm_gtk = shared_library(
   c_args: cflags,
   link_args: ldflags,
   link_depends: symbol_map,
+  link_whole: libwireless_security_libnm_glib,
   install: true,
   install_dir: nma_libdir
 )
diff --git a/src/libnma/meson.build b/src/libnma/meson.build
index 0659268a375c45cc..98a72aa733c84c50 100644
--- a/src/libnma/meson.build
+++ b/src/libnma/meson.build
@@ -47,15 +47,15 @@ incs = [
   top_inc,
   shared_inc,
   src_inc,
-  libnma_inc
+  libnma_inc,
+  wireless_security_inc
 ]
 
 deps = [
   gtk_dep,
   gudev_dep,
   libnm_dep,
   libutils_libnm_dep,
-  libwireless_security_libnm_dep
 ]
 
 cflags = [
@@ -102,6 +102,7 @@ libnma = shared_library(
   c_args: cflags,
   link_args: ldflags,
   link_depends: symbol_map,
+  link_whole: libwireless_security_libnm,
   install: true,
   install_dir: nma_libdir
 )
diff --git a/src/meson.build b/src/meson.build
index c8e287237fd128f0..45ff246352ba53d3 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -61,13 +61,20 @@ sources += gnome.compile_resources(
   dependencies: resource_data
 )
 
+incs = [
+  top_inc,
+  utils_inc,
+  src_inc,
+  shared_inc,
+  wireless_security_inc
+]
+
 deps = [
   gtk_dep,
   libnm_dep,
   libnma_dep,
   libnotify_dep,
   libsecret_dep,
-  libwireless_security_libnm_dep,
   m_dep
 ]
 
@@ -109,6 +116,7 @@ executable(
   c_args: cflags,
   link_args: ldflags,
   link_depends: linker_script_ver,
+  link_whole: libwireless_security_libnm,
   install: true,
   install_dir: nma_bindir
 )
diff --git a/src/wireless-security/meson.build b/src/wireless-security/meson.build
index e3efcdb4ca0a39d8..a4fbe97d36ae9af7 100644
--- a/src/wireless-security/meson.build
+++ b/src/wireless-security/meson.build
@@ -57,30 +57,20 @@ libwireless_security_libnm = static_library(
   dependencies: deps
 )
 
-libwireless_security_libnm_dep = declare_dependency(
-  link_with: libwireless_security_libnm,
-  include_directories: wireless_security_inc
-)
-
 if enable_libnm_gtk
   deps = [
     gtk_dep,
     libnm_glib_dep,
     libutils_libnm_glib_dep
   ]
 
   cflags = '-DNETWORKMANAGER_COMPILATION=NM_NETWORKMANAGER_COMPILATION_LIB_LEGACY'
 
   libwireless_security_libnm_glib = static_library(
     'wireless-security-libnm-glib',
     sources: sources,
     include_directories: incs,
     dependencies: deps,
     c_args: cflags
   )
-
-  libwireless_security_libnm_glib_dep = declare_dependency(
-    link_with: libwireless_security_libnm_glib,
-    include_directories: wireless_security_inc
-  )
 endif
-- 
2.15.0

